targeted/WordPress User Enumeration.js

```diff
@@ -0,0 +1,204 @@
+            url + "/wp-json/wp/v2/users"
+        );
+    }
+
+    if (archiveAuthors.length) {
+        logger("Vulnerable to user enumeration.");
+        var alertEvidence = "Found author(s) -> ";
+        for (var i in archiveAuthors) {
+            alertEvidence += archiveAuthors[i].username + ", ";
+        }
+        url = url + "/?author=" + archiveAuthors[0].iterator;
+        // Raise alert
+        raiseAlert(
+            pluginid,
+            3, // risk: 0: info, 1: low, 2: medium, 3: high
+            3, // confidence: 0: falsePositive, 1: low, 2: medium, 3: high, 4: confirmed
+            alertName,
+            alertDesc,
+            alertEvidence,
+            alertSol,
+            alertReference,
+            cweId,
+            wascId,
+            msg,
+            url
+        );
+    }
+    logger("Script run completed.");
+}
+
+/**
+ * 
+ * @param {HttpMessage} msg 
+ * @returns {Object.object}
+ */
+function archiveJson(msg) {
+    logger("Testing feed enumeration");
+    var authors = [];
+    var newReq = sendReq(msg, '/wp-json/wp/v2/users');
+    var responseHeader = newReq.getResponseHeader();
+    if (responseHeader.getStatusCode() === 200 && responseHeader.getHeader("Content-Type").contains('json')); {
+        var responseBody = JSON.parse(newReq.getResponseBody().toString());
+        logger("200 response with JSON");
+        for (var i in responseBody) {
+            authors.push(responseBody[i].slug); // Usernames (author names)
+        }
+    }
+    return authors;
+}
+
+/**
+ * 
+ * @param {HttpMessage} msg - The HttpMessage Object
+ * @returns {Object.object}
+ */
+function userEnumerate(msg) {
+    logger("Testing author enumeration");
+    // Array to store found usernames
+    var authors = [];
+    // Initialise iterator
+    var i = 1;
+    // To iterate first 10 possible enumerations
+    while (i && i < 10) {
+        var newReq = sendReq(msg, i);
+        // If enum possible, 301 status code is constant
+        if (newReq.getResponseHeader().getStatusCode() === 301) {
+            // Get the redirection location
+            var redirect = newReq.getResponseHeader().getHeader("Location");
+            // Extract username from Redirect Location
+            try {
+                var authorName = redirect.split("author/")[1].split("/")[0];
+                var authorEntry = {
+                    username: authorName,
+                    iterator: i
+                };
+                authors.push(authorEntry);
+                logger("Vulnerable Endpoint.");
+                i++;
+            } catch (err) {
+                // If redirection but no username found
+                if (err instanceof TypeError) {
+                    logger("Endpoint not vulnerable.");
+                    i++;
+                }
+            }
+        } else {
+            i++;
+        }
+    }
+    return authors;
+}
+
+/**
+ * 
+ * @param {HttpMessage} msg 
+ * @param {Int} query 
+ * @returns 
+ */
```

Comment:
This jsdoc is kind of pointless if the param/returns aren't going to actually be further defined.